### ISSUE TITLE
op_global_planner cannot read the vectormap lane data 

### ISSUE THREAD

**Issue by ROS-lover** [autocore-ai/MapToolbox#14]
***Updated with the following zip file containing the scene *.unity as well as the output vectormap.***
[vect_unity_test.zip](https://github.com/autocore-ai/MapToolbox/files/3444552/vect_unity_test.zip)

Hello,

I have used this plugin on Unity and it worked for most of time time. However I cannot understand why these maps attached which are just lane data and differ from each other only slightly, won't work with op_global planner in autoware. map folder 19 and 23 work but not the rest. They only differ slightly from each other.

[drive-download-20190727T233322Z-001.zip](https://github.com/autocore-ai/MapToolbox/files/3438700/drive-download-20190727T233322Z-001.zip)

Terminal Error

process[op_global_planner-1]: started with pid [6166]
all processes on machine have died, roslaunch will exit
shutting down processing monitor...
... shutting down processing monitor complete
done
['roslaunch', 'op_global_planner', 'op_global_planner.launch', 'enableLaneChange:=False', 'enableReplan:=True', 'enableSmoothing:=True', 'enableRvizInput:=True', 'pathDensity:=1', 'velocitySource:=1']
pid=6539
sched policy=OTHER prio=0
Received Lanes704
Received Points1089
Received dtLanes739
Received Lines342
Received Nodes739
Map Version 2
>> Extracting Lanes ...
>> Fix Waypoints errors ...
>> Create Missing lane connections ...
terminate called after throwing an instance of 'std::out_of_range'
what():  vector::_M_range_check: __n (which is 0) >= this->size() (which is 0)
[op_global_planner-1] process has died [pid 6558, exit code -6, cmd /home/ocp/autoware/ros/install/op_global_planner/lib/op_global_planner/op_global_planner __name:=op_global_planner __log:=/home/ocp/.ros/log/76e8791a-afcc-11e9-b129-a4fc771a0d0d/op_global_planner-1.log].
log file: /home/ocp/.ros/log/76e8791a-afcc-11e9-b129-a4fc771a0d0d/op_global_planner-1*.log
... logging to /home/ocp/.ros/log/76e8791a-afcc-11e9-b129-a4fc771a0d0d/roslaunch-ocp-6539.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

Thanks in advance for looking into it!

**Comment by MoeLang** [comment]
@ROS-lover Can you provide the scene file (named by *.unity) with this problem ?

**Comment by ROS-lover** [comment]
@MoeLang Thanks. Just updated the issue's attachment with the file requested placed in the vect_unity_test.zip file. Thank you!

**Comment by MoeLang** [comment]
@ROS-lover Thank you! From the scene file I can't find obvious problems, it seems caused by precision lost during serialization and I'll dig into open planner.

**Comment by ROS-lover** [comment]
@ROS-lover That's right, I also was quite surprised since I did not find any obvious reason in format of the file generated, is it possible it has to do with something related to open planner?

**Comment by MoeLang** [comment]
@ROS-lover Yes, If open planner works wrong,  I should provide some fix. 
Currently all tests and fixes are aimed at Autoware ver 1.12

**Comment by ROS-lover** [comment]
@MoeLang , thanks, I can switch to that version if the fix provided works.

**Comment by MoeLang** [comment]
@ROS-lover Try this fix, it works well on my machines.
![image](https://user-images.githubusercontent.com/7805397/62281473-6849ee80-b480-11e9-861b-18d3ac700b06.png)

**Comment by ROS-lover** [comment]
@MoeLang Thanks Moe, what you suggested works just fine with the input vectormap I have attached. I will try to see if I face a problem and can let you know. Thanks once again!

**Comment by Fangwq** [comment]
@ROS-lover  Hi，I follow the your procedure and get the result. The op_global_planner process doesn't died and the vector map doesn't [mix up](https://github.com/autowarefoundation/autoware_ai/issues/712). However, the point map and vector map doesn't match. How to solve this problem? Thank you very much! I run it with autoware and lgsvl simulator(shalun map).  In addition, when I operate the car in lgsvl simulator, the points raw, and points map also doesn't match any more.
![image](https://user-images.githubusercontent.com/19931365/65310791-ada5b500-dbc1-11e9-911c-fd038c94a58c.png)

**Comment by ROS-lover** [comment]
Hi @Fangwq. The issue I posed was specifically for a vectormap i prepared regardless of an actual location, and without pointcloud map. So it's normal they pcd map and vector map dont match since they are not made on top of one another. About the open planner, I used an older version of autoware which needed a line of code to change suggested by Moe and looks like in the newer version it's not a problem. hope this addressed your concern.
